### DISCUSSION TITLE
Error when updating the public repo

### DISCUSSION THREAD

**Discussion by mgutierreztudelft** [tudelft/AE4353-Y24#15] [General]
Hello,

When I update the public repo by introducing "git fetch upstream
git rebase upstream/main" in the wsl2, an error occurs: error: cannot rebase: "You have unstaged changes.
error: Please commit or stash them."

I tried by introducing git rebase --autostash upstream/main but another error followed: "error: The following untracked working tree files would be overwritten by reset --hard:
        data/.gitkeep
Please move or remove them before you reset --hard.
Aborting
fatal: could not reset --hard"
Anything that would help update the github repo would be appreciated.
Thanks

**Comment by yilun-wu** [comment]
As the error message suggests, you should revert any changes to the files tracked by git which includes `data/.gitkeep`. When we mount the dataset folder into devcontainer, we overwrited the original data folder in the git repo so that `data/.gitkeep` is deleted. Now we need to bring it back by `git restore data/.gitkeep`. After this, you can continue with your git rebase command.

**Reply by kibangadas** [reply]
I have the exact same problem and restoring data/.gitkeep didn't help. I get the following error:
`error: cannot rebase: You have unstaged changes.
error: additionally, your index contains uncommitted changes.
error: Please commit or stash them.`
How can I fix this?

**Reply by yilun-wu** [reply]
As the error message suggests, you should commit your already staged changes (if you plan to do so) or stash them (if you are not planning to commit just yet). Once the repository is free of any unstaged and staged changes, your rebase should work.
Perhaps the following videos would help: 
https://www.youtube.com/watch?v=DkWDHzmMvyg
https://www.youtube.com/watch?v=lH3ZkwbVp5E
